oceannavigator/frontend/src/components/Map.jsx

```diff
@@ -1202,18 +1264,25 @@ export default class Map extends React.PureComponent {
 
     datalayer.setSource(newSource);
 
-    // Update colour scale
-    if (this.scaleViewer != null) {
-      this.map.removeControl(this.scaleViewer);
+    if (this.shouldUpdateScaleViewer(this.props.state.dataset,
```

Comment:
Added a shouldUpdate check here since otherwise we'd be reconstructing the variable scale bar every time the component updates which is silly. We only need to reconstruct when the dataset, variable, or actual variable scale values change (0, 30 for example).

Checked it and it works fine...it does disappear for a little bit when changing variables or datasets but that's a result of crappy javascript architecture.